src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -1158,7 +1158,8 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
                 if (pResult->exactContextNeedsRuntimeLookup &&
                     pResolvedToken.tokenContext == contextFromMethodBeingCompiled() &&
                     constrainedType == null &&
-                    exactType == MethodBeingCompiled.OwningType)
+                    exactType == MethodBeingCompiled.OwningType &&
+                    !MethodBeingCompiled.HasInstantiation)
```

Comment:
It may be worth a comment that this condition is required to make dictionaries pre-computed by the scanner work.

Reply:
(The condition is not present and needed in the R2R version of this method and it is not obvious why.)